### DISCUSSION TITLE
Action action on Page does not execute function

### DISCUSSION THREAD

**Discussion by calimadevteam** [filamentphp/filament#4015] [Help]
Hi!
I have a custom Page that has Actions but they do not seem to execute the functions that are described in their  'action'.

Here is the code:

```
class documentRevisionsCompare extends Page 
{
    protected static string $resource = DocumentResource::class;

    protected static string $view = 'filament.resources.document-resource.pages.document-revisions-compare';

    protected function getActions(): array{
        return [        
            Action::make('restore')
                ->action('testFunction')
                ->requiresConfirmation(),
            Action::make('cancel')
                ->url('/admin/documents/' . $this->document->id.'/edit')
                ->color('secondary')            
        ];
    }

    public function testFunction(): void{
        $this->notify('danger', 'Please select revisions.');
     }
}
```
The second Action that only has a 'url' works fine tho. Any ideas on what can be the problem?

**Comment by danharrin** [comment]
Tried to see if a dd() will run? If not, try passing a callback function to the action instead of a public method?

**Reply by calimadevteam** [reply]
I tried both, but unfortunately, they didn't work.

**Reply by danharrin** [reply]
Inspect the button that runs the action. What does the HTML look like?

**Reply by calimadevteam** [reply]
It is basically empty. It just has the filament page tag in it:
``
<x-filament::page></x-filament::page>
``

**Reply by danharrin** [reply]
No, I mean the `<button>`, send the HTML for that from the browser

**Reply by calimadevteam** [reply]
![image](https://user-images.githubusercontent.com/99034512/190394470-6983ae00-d542-4f30-9bde-05cbbfa76b2b.png)

**Reply by danharrin** [reply]
Not sure... a modal should open when you click that 🤷‍♂️
